docs/source/decisions/0004-versioning.rst

```diff
@@ -0,0 +1,61 @@
+########################
+CWMS Data Api Versioning
+########################
+
+
+Summary
+=======
+
+Maintaining backwards compatibility while improving future difficulty has proven sufficiently difficulty that change
+is required.
+
+The API as a whole will retain the calendar based versioning for formal releases.
+Data *SHOULD* be versioned, if appropriate/needed, with otherwise backwards compatible changes to query parameters.
+Endpoints will be places under a new "api version" for backwards incompatible or confusing parameter changes.
+
+e.g.
+
+`https://host/cwms-data/locations`
+
+can become
+
+`https://host/cwms-data/v2/locations`
+
+As additional endpoints require such a change they should be added to an existing increased version.
+
+Example:
+
+given above and a v1 `timeseries` and yet another `locations` improvements
+
+.. code-block:: bash
+
+    # new time series becomes
+    cwms-data/v2/timeseries
+    # the new location becomes
+    cwms-data/v3/locations
+
+.. NOTE::
+
+    Or is that confusing and we should just allows add a new endpoint to the highest endpoint version?
```

Comment:
Worth calling out how various methods would get versioned with path versioning:

I presume we would want `cwms-data/v3/locations GET`, `cwms-data/v3/locations/<name> GET`, `cwms-data/v3/locations/<name> DELETE`, `cwms-data/v3/locations/<name> POST` would all get updated to the v3 even if only the `cwms-data/v3/locations/<name> GET` became backwards incompatible?